src/test/lrsql/admin/protocol_test.clj

```diff
@@ -91,176 +91,177 @@
-    (testing "Admin account authentication"
-      (is (-> (adp/-authenticate-account lrs test-username test-password)
-              :result
-              uuid?))
-      (is (-> (adp/-authenticate-account lrs test-username "badPass")
-              :result
-              (= :lrsql.admin/invalid-password-error)))
-      (is (-> (adp/-authenticate-account lrs "foo" "bar")
-              :result
-              (= :lrsql.admin/missing-account-error))))
-    (testing "Admin account existence check"
-      (let [account-id (-> (adp/-authenticate-account lrs
-                                                      test-username
-                                                      test-password)
-                           :result)]
-        (is (adp/-existing-account? lrs account-id)))
-      (let [bad-account-id #uuid "00000000-0000-4000-8000-000000000000"]
-        (is (not (adp/-existing-account? lrs bad-account-id)))))
-    (testing "Admin password update"
-      (let [account-id   (-> (adp/-authenticate-account lrs
+    (try
+      (testing "Admin account insertion"
+        (is (-> (adp/-create-account lrs test-username test-password)
+                :result
+                uuid?))
+        (is (-> (adp/-create-account lrs test-username test-password)
+                :result
+                (= :lrsql.admin/existing-account-error))))
+      (testing "Admin account get"
+        (let [accounts (adp/-get-accounts lrs)]
+          (is (vector? accounts))
+          (is (some (fn [acct] (= (:username acct) test-username)) accounts))))
+      (testing "Admin account authentication"
+        (is (-> (adp/-authenticate-account lrs test-username test-password)
+                :result
+                uuid?))
+        (is (-> (adp/-authenticate-account lrs test-username "badPass")
+                :result
+                (= :lrsql.admin/invalid-password-error)))
+        (is (-> (adp/-authenticate-account lrs "foo" "bar")
+                :result
+                (= :lrsql.admin/missing-account-error))))
+      (testing "Admin account existence check"
+        (let [account-id (-> (adp/-authenticate-account lrs
                                                         test-username
                                                         test-password)
-                             :result)
-            new-password "iLoveNoSql"]
-        (testing "Valid update"
-          (adp/-update-admin-password lrs account-id test-password new-password)
-          (is (-> (adp/-authenticate-account lrs
-                                             test-username
-                                             new-password)
+                             :result)]
+          (is (adp/-existing-account? lrs account-id)))
+        (let [bad-account-id #uuid "00000000-0000-4000-8000-000000000000"]
+          (is (not (adp/-existing-account? lrs bad-account-id)))))
+      (testing "Admin password update"
```

Comment:
Same as above with password update

Reply:
I don't think there is a specific one for that, I'll work one up

Reply:
And unlike with above, it _doesn't_ seem like you have tests for bad password updates in the route tests.

Reply:
48c8f0e1aad46d277e7db30decda00fe5710b493